src/ValueHandler/ChannelPricingValueHandler.php

```diff
@@ -85,11 +90,27 @@ public function handle($subject, string $attribute, array $value): void
                     $channelPricing = $this->channelPricingFactory->createNew();
                     $channelPricing->setChannelCode($channel->getCode());
                 }
-                $channelPricing->setPrice((int) round($price * 100));
+                $setterMethod = $this->getPriceSetterMethod();
+                $channelPricing->$setterMethod((int) round($price * 100));
                 if ($isNewChannelPricing) {
                     $subject->addChannelPricing($channelPricing);
                 }
             }
         }
     }
+
+    private function getPriceSetterMethod(): string
+    {
+        switch ($this->syliusPropertyPath) {
+            case 'price':
+                return 'setPrice';
+            case 'original_price':
+                return 'setOriginalPrice';
+            default:
+                throw new \InvalidArgumentException(sprintf(
+                    "Sylius property path not allowed. Expected 'price' or 'original_price', received: %s",
+                    $this->syliusPropertyPath
+                ));
+        }
+    }
```

Comment:
First of all, I'd use the [Symfony Property Access Component](https://symfony.com/doc/current/components/property_access.html) to set the object value as already done in `GenericPropertyValueHandler` or `TranslatablePropertyValueHandler`.

Second, I'd not check if the property path is `price` or `original_price`. In some projects the channel pricing entity could have been extended with other properties. It will be the property access component to throw an exception if the given property path is not valid.